functions-and-operators/string-functions.md

```diff
@@ -107,19 +107,128 @@ Return a string containing binary representation of a number.
 
 ### [`BIT_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bit-length)
 
-Return length of argument in bits.
+The `BIT_LENGTH()` function is used to return the lenght of a given argument in bits. It calculates the length of the argument and then returns the result in bits.
+
+Examples:
+
+```sql
+SELECT BIT_LENGTH("TiDB");
++----------+
+| BIT_LENGTH("TiDB") |
++----------+
+| 8 (T) + 8 (i) + 8 (D) + 8 (B) = 40 bits |
++----------+
+```
+
+```sql
+SELECT CustomerName, BIT_LENGTH(CustomerName) AS BitLengthOfName FROM Customers;
+| CustomerName|BitLengthOfName |
+|-------------|----------------|
+| Albert Einstein | 120 bits |
+| Robert Oppenheimer | 144 bits |
+```
+
+```sql
+SELECT BIT_LENGTH("PingCap 123")
++----------+
+| BIT_LENGTH("PingCap 123") |
++----------+
+| 8 (P) + 8 (i) + 8 (n) + 8 (g) + 8 (C) + 8 (a) + 8 (p) + 8 () + 8 (1) + 8 (2) + 8 (3) = 88 bits |
++----------+
+```
+
+> **Note:**
+>
+> The second example operates under the assumption that there is a database with a record titled `Customers` and a field inside titled `CustomerName`
 
 ### [`CHAR()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char)
 
-Return the character for each integer passed.
+The `CHAR()` function is used to get the corresponding character value of a specific ASCII value. It performs the opposite function of `ASCII()`. Unlike `ASCII()`, which can get the ASCII value of a specific character, `CHAR()` can get the character of a specific ASCII value.
+
+Examples:
+
+```sql
+SELECT CHAR(65);
+
++------------+
+|  CHAR(65) |
++------------+
+|        A |
++------------+
+```
+
+```sql
+SELECT CHAR(84);
+
++------------+
+|  CHAR(84) |
++------------+
+|        T |
++------------+
+```
+
+> **Note:**
+>
+> The `CHAR()` function can also be used to get the corresponding character values of ASCII values that extend beyond the standard ASCII range (0 - 127). It can also get the corresponding character value of a unicode value.
+
+Further Examples:
+
+```sql
+/*For extended ASCII: */
+
+SELECT CHAR(128);
+
++------------+
+|  CHAR(128) |
++------------+
+|       Ç |
++------------+
+```
+
+```sql
+/* For Unicode: */
+
+SELECT CHAR (233);
```

Comment:
```suggestion
SELECT CHAR(233);
```

Reply:
233 is the right value for `é` in ISO-8859-15, but not for UTF-8.

From the `iso_8859-15(7)` manpage:
```
       Oct   Dec   Hex   Char   Description
       351   233   E9     é     LATIN SMALL LETTER E WITH ACUTE
```

In Python:
```
>>> 'é'.encode('iso-8859-15')
b'\xe9'
>>> 'é'.encode('utf-8')
b'\xc3\xa9'
>>> int("0xe9", 0)
233
>>> int("0xc3a9", 0)
50089
```

`mysql --skip-binary-as-hex ...` gives this with a unicode terminal:
```
mysql> SELECT CHAR(50089);
+-------------+
| CHAR(50089) |
+-------------+
| é           |
+-------------+
1 row in set (0.00 sec)
```